### PR TITLE
GAWB-1351

### PR DESCRIPTION
**Pull request by skwalker**
Orch PR: https://github.com/broadinstitute/firecloud-orchestration/pull/342
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs [hunk 1]
```diff
@@ -181,6 +179,12 @@
    :component-did-mount
    (fn [{:keys [props state refs]}]
      (let [{:keys [library-schema questions]} props]
+        (let [versions (get-in library-schema [:versions])]
+          (mapv (fn [version]
+                  (let [currentVersion (get-in library-schema [:properties (keyword version) :current])]
+                    (utils/log currentVersion)
```
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs]: log statement

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs [hunk 2]
```diff
@@ -181,6 +179,12 @@
...
+                    (swap! state update :attributes assoc (keyword version) currentVersion))) ;;js/parseInt should work?
```
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs]: Do you want the comment here? I suggest fix or remove.

**Comment by rushtong** [comment]
If the version numbers are not in the wizard, then how are their values updated? The way the original code worked, whenever data values were saved by the user, so would the lmsvn number be updated.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs (hunk 2 above)
Comment (davidangb) [reply src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs]: actually ... this entire `let` block. Are these version numbers anywhere? (they could be, I'm just not seeing it)

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/library_view.cljs [hunk 3]
```diff
@@ -15,21 +15,17 @@
         (common/attribute-list? value) (clojure.string/join ", " (common/attribute-values value))
         :else value))
 
-(defn- render-library-row [key value doid]
+(defn- render-library-row [key value]
   (when value
     [:div {:style {:display "flex" :padding "0.5em 0" :borderBottom (str "2px solid " (:line-default style/colors))}}
      [:div {:style {:flexBasis "33%" :fontWeight "bold" :paddingRight "2em"}} key]
-     [:div {:style {:flexBasis "67%"}} value
-      (if (not (nil? doid)) [:span {:style {:fontStyle "italic"}} (str " (" doid ")")])]]))
```
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/library_view.cljs:23]: are you intentionally removing the ontology stuff?

Comment (skwalker) [reply src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/library_view.cljs:23]: yeah that's intentional (it wasn't actually doing anything anyways)

**Comment by rushtong** [comment]
By changing the semantics around `lmsvn` and `editable`, it shows up in the UI now as something that the user has to enter. I'm pretty sure this is incorrect. It needs to be there as an un-editable field that is automatically populated with the most current version. 

screen shot 2017-01-06 at 1 39 56 pm

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs (hunk 2 above)
Comment (skwalker) [reply src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/catalog.cljs]: Version numbers are hidden when you're cataloging, getting saved behind the scenes, and then viewable in dataset attributes --> unless it's never been catalogued with a version # before. I also found a hack to get it to save properly as a number... our validation code is a little messed up

**Comment by skwalker** [comment]
@rushtong Dataset Version is actually something else, it's not related to lmsvn or dulvn.

**Comment by rushtong** [comment]
@skwalker - Yes, you're right - ignore my lmsvn comments - it is getting saved correctly.

**Comment by davidangb** [comment]
👍 as TL (I did not pull and test)

**Review by davidangb — APPROVED** [review]

**Comment by rushtong** [comment]
👍  
I did test this locally.

**Pull request merged**